File: 278-first-bad-version/278-first-bad-version.py
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return an integer
# def isBadVersion(version):

class Solution:
    def firstBadVersion(self, n):
        s, l = 0, n
        flag = -1
        while s<=l:
            mid = (s+l)//2
            if isBadVersion(mid) == False:
                s = mid + 1
            else:
                flag = mid
                l = mid - 1
        return flag
                

File: 278-first-bad-version/278-first-bad-version.go
/** 
 * Forward declaration of isBadVersion API.
 * @param   version   your guess about first bad version
 * @return 	 	      true if current version is bad 
 *			          false if current version is good
 * func isBadVersion(version int) bool;
 */

func firstBadVersion(n int) int {
    l, h := 0, n
    var mid int
    var flag int
    for l<=h{
        mid = (l+h)/2
        if isBadVersion(mid) == false{
            l = mid + 1
        }else{
            flag = mid
            h = mid - 1
        }
    }
    return flag
}